Instruments/include/DYFileDescriptor.h

```diff
@@ -0,0 +1,125 @@
+/*! Definition of the file descriptor.
+This file is part of https://github.com/hh-italian-group/hh-bbtautau. */
+
+#pragma once
+
+#include <list>
+#include <istream>
+#include <ostream>
+#include "AnalysisTools/Core/include/EnumNameMap.h"
+#include "AnalysisTools/Core/include/NumericPrimitives.h"
+#include "AnalysisTools/Core/include/PhysicalValue.h"
+#include "h-tautau/Analysis/include/SummaryTuple.h"
+#include "AnalysisTools/Core/include/ConfigReader.h"
+
+namespace analysis {
+
+namespace sample_merging{
+
+enum class FileType { inclusive, exclusive };
+ENUM_NAMES(FileType) = {
+    { FileType::inclusive, "inclusive" },
+    { FileType::exclusive, "exclusive" }
+};
+
+struct DYBinDescriptor {
+    std::string name;
+    std::string file_path;
+    FileType fileType;
+    analysis::Range<int> n_jet;
+    analysis::Range<int> n_bjet;
+    analysis::Range<int> n_ht;
+
+    PhysicalValue nu;
+    PhysicalValue weight;
+    std::string ref_sample;
+    FileType ref_fileType;
+
+    DYBinDescriptor()
+        : n_jet(0,0), n_bjet(0,0),n_ht(0,0),nu(0.0, std::numeric_limits<double>::infinity()),
+          weight(std::numeric_limits<double>::quiet_NaN()) {}
+
+    bool Contains(const ntuple::GenId& genId) const
+    {
+        return n_jet.Contains(genId.n_partons) &&
+              n_bjet.Contains(genId.n_b_partons) &&
+              n_ht.Contains(genId.ht10_bin);
+    }
+
+    static std::vector<DYBinDescriptor> LoadConfig(const std::string& config_file)
+    {
+        std::vector<DYBinDescriptor> dyBinDescriptors;
+        dyBinDescriptors.clear();
+        std::ifstream cfg(config_file);
+        while (cfg.good()) {
+            std::string cfgLine;
+            std::getline(cfg,cfgLine);
+            if (!cfgLine.size() || cfgLine.at(0) == '#') continue;
+            auto columns = ConfigEntryReader::ParseOrderedParameterList(cfgLine, true);
+            std::istringstream ss(cfgLine);
+            DYBinDescriptor descriptor;
+            if(columns.size() >= 6)
+                ss >> descriptor.n_jet >> descriptor.n_bjet >> descriptor.n_ht;
+            if (columns.size() >= 11){
+                double col_weight = analysis::Parse<double>(columns.at(6));
+                double col_weight_err = analysis::Parse<double>(columns.at(7))*col_weight;
+                analysis::PhysicalValue weight(col_weight,col_weight_err);
+                descriptor.weight = weight;
```

Comment:
can do directly:
descriptor.weight = PhysicalValue(col_weight,col_weight_err)
same for nu.